### ISSUE TITLE
Difficulty in Implementing Model Re-Training for Time-Series Data with Pre-Trained Model

### ISSUE THREAD

**Issue by AyushBhardwaj321** [unit8co/darts#2351]
Hello everyone,

I'm encountering an issue with implementing incremental learning for time-series data using a pre-trained model, and I'm seeking guidance on how to address it.

**Description:**
I have a substantial dataset consisting of 2.5 million data points, structured as follows:

```
                                ds    y
0        2020-10-21 12:57:47+00:00  0.0
1        2020-10-21 12:57:48+00:00  0.0
2        2020-10-21 12:57:49+00:00  0.0
3        2020-10-21 12:57:50+00:00  0.0
4        2020-10-21 12:57:51+00:00  0.0
...                            ...  ...
2591996  2020-11-20 12:57:43+00:00  7.0
2591997  2020-11-20 12:57:44+00:00  7.0
2591998  2020-11-20 12:57:45+00:00  7.0
2591999  2020-11-20 12:57:46+00:00  6.0
2592000  2020-11-20 12:57:47+00:00  6.0

[2592001 rows x 2 columns]
```
**Objective:**
My aim is to train an initial model, let's call it **Model_v1**, on this historical data. However, due to the large size of the dataset, training the model from scratch is time-consuming. Therefore, I want to explore the possibility of using **Model_v1** as a base model and incrementally updating it with new data obtained from a real-time source, which arrives approximately every minute. Ideally, I would like to re-train the model on a weekly or monthly basis to adapt to any changes in the underlying patterns.

**Challenge:**
Although I've managed to save the historical model (**Model_v1**), I'm encountering difficulties when attempting to load it and fit new data onto it. The incremental learning process seems to be failing, and I'm unsure about the correct approach to achieve this.

**Request for Assistance:**
I would greatly appreciate any insights, suggestions, or best practices on how to effectively implement incremental learning with pre-trained models for time-series data. Additionally, if anyone has encountered similar challenges or has successfully implemented such a solution, your input would be invaluable.

Thank you in advance for your assistance and guidance.

**Comment by madtoinou** [comment]
Hi @AyushBhardwaj321,

You can find an example on re-training/fine-tuning pretrained model [here](https://unit8co.github.io/darts/userguide/torch_forecasting_models.html#re-training-or-fine-tuning-a-pre-trained-model). We recommend using the method `load_weights_from_checkpoint()` as it provides a lot of control over the parameters used for the second training session.

**Comment by dennisbader** [comment]
And `load_weights()` from a manual save (model saved with `model.save()`).

**Comment by madtoinou** [comment]
1. Yes, only the neural-network based models support this approach. You can find the list [here](https://unit8co.github.io/darts/README.html#forecasting-models), in the "PyTorch (Lightning)-based Models" section.
2. No, these models do not support fine-tuning, they are retrained from scratch every-time you call `fit()`; it's one of the inconvenient of regression models.

**Comment by dennisbader** [comment]
@AyushBhardwaj321, fine-tuning only works with our neural network models as @madtoinou pointed out. There you can also adapt the learning rate and other parameters for fine-tuning so your model won't overfit on the data used for fine-tuning.

**Comment by madtoinou** [comment]
Hi @AyushBhardwaj321,

I don't think that the documentation includes such a tutorial but you could easily combine the code snippet mentioned above to load weights from a checkpoint with any of the other example notebook such as the [quickstart](https://unit8co.github.io/darts/quickstart/00-quickstart.html#Machine-learning-and-global-models), [this one](https://unit8co.github.io/darts/examples/01-multi-time-series-and-covariates.html) or [this one](https://unit8co.github.io/darts/examples/21-TSMixer-examples.html).

I would recommend checking the various examples, I am sure that you will find something very close to what you want to achieve and use it as a base.
